src/coreclr/jit/hwintrinsicarm64.cpp

```diff
@@ -686,12 +686,40 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
 
             IntrinsicNodeBuilder nodeBuilder(getAllocator(CMK_ASTNode), sig->numArgs);
 
+            // TODO-CQ: We don't handle contiguous args for anything except TYP_FLOAT today
+
+            GenTree* prevArg           = nullptr;
+            bool     areArgsContiguous = (simdBaseType == TYP_FLOAT);
```

Comment:
The existing handling in `areArgumentsContiguous` and elsewhere only checks and supports `TYP_FLOAT` today.

It shouldn't be difficult to extend to other regular types (`TYP_INT`, `TYP_UINT`, `TYP_LONG`, `TYP_ULONG`, and `TYP_DOUBLE`). It might require a bit more work to extend to small types (`TYP_BYTE`, `TYP_UBYTE`, `TYP_SHORT`, and `TYP_USHORT`).

However, I'm leaving it to a follow up PR to minimize the churn here.